src/main/java/jenkins/scm/api/SCMHead.java

```diff
@@ -155,47 +146,34 @@ public String toString() {
     }
 
     /**
-     * Gets all actions used to decorate the behavior of this branch.
-     * May be overridden to create a new list, perhaps with additions.
-     * @return a list of all actions associated with this branch (by default, an unmodifiable list searching {@link TransientActionFactory}s)
-     * @see Actionable#getAllActions
+     * Returns an empty list.
+     * @return an empty list
      * @since 1.1
+     * @deprecated this was added to the API in error. Retained for backwards binary compatibility only. Use
+     * {@link SCMSource#fetchActions(SCMHead, SCMHeadEvent, TaskListener)} to get the actions associated with a
+     * {@link SCMHead}
      */
+    @Restricted(DoNotUse.class)
+    @Deprecated
     @NonNull
-    @Exported(name="actions")
     public List<? extends Action> getAllActions() {
-        List<Action> actions = new ArrayList<Action>();
-        for (TransientActionFactory<?> taf : ExtensionList.lookup(TransientActionFactory.class)) {
-            if (taf.type().isInstance(this)) {
-                try {
-                    actions.addAll(createFor(taf));
-                } catch (Exception e) {
-                    LOGGER.log(Level.SEVERE, "Could not load actions from " + taf + " for " + this, e);
-                }
-            }
-        }
-        return Collections.unmodifiableList(actions);
-    }
-    private <T> Collection<? extends Action> createFor(TransientActionFactory<T> taf) {
-        return taf.createFor(taf.type().cast(this));
+        return Collections.emptyList();
     }
 
     /**
-     * Gets a specific action used to decorate the behavior of this branch.
-     * May be overridden but suffices to override {@link #getAllActions}.
+     * Returns {@code null}.
      * @param <T> a desired action type to query, such as {@link ChangeRequestAction}
      * @param type type token
-     * @return an instance of that action interface (by default, filters {@link #getAllActions})
-     * @see Actionable#getAction(Class)
+     * @return {@code null}
      * @since 1.1
+     * @deprecated this was added to the API in error. Retained for backwards binary compatibility only. Use
+     * {@link SCMSource#fetchActions(SCMHead, SCMHeadEvent, TaskListener)} to get the actions associated with a
+     * {@link SCMHead}
      */
+    @Restricted(DoNotUse.class)
+    @Deprecated
     @CheckForNull
     public <T extends Action> T getAction(@NonNull Class<T> type) {
-        for (Action action : getAllActions()) {
-            if (type.isInstance(action)) {
-                return type.cast(action);
-            }
-        }
         return null;
```

Comment:
At least maintain compatibility for callers of `ChangeRequestAction`: if `this instanceof ChangeRequestSCMHead`, return an action delegating to the new APIs. That should allow current `blueocean-pipeline-api-impl` releases to work.